### PR TITLE
framework: Add active polling for acquiring policy file

### PR DESCRIPTION
**Pull request by vmojzis**
setroubleshoot server crashes when the policy file is used by other
process. Include 10s active polling for the policy file.

In python 2 (swig) failure to open /sys/fs/selinux/policy results in
ValueError "Device or resource busy", while in python 3 the same
ValueError causes SystemError (the ValueError is saved in __context__).

Resolves: rhbz#1583241

### CODE REVIEW

**Comment by vmojzis** [comment]
Add error message when polling fails.

**Comment by vmojzis** [comment]
Updated to python3-only code. Added global variable to stop polling, because sys.exit() does not work in signal handler.

**Comment by vmojzis** [comment]
Updated

File: framework/src/setroubleshoot/server.py [hunk 1]
```diff
@@ -115,6 +117,12 @@ def sighandler(signum, frame):
         return
 
 
+def polling_failled_handler(signum, frame):
+    log_debug("received signal=%s" % signum)
+    global continue_polling
+    continue_polling = False
+
```
Comment (bachradsusi) [inline framework/src/setroubleshoot/server.py]: I'd rather see the same concept as is in the stable branch.
Given the problem in `selinux.audit2why`, `sys.exit()` doesn't seem to work in this case, but you can temporary force exit using `os._exit()`. Please also a FIXME comment with some explanation and reference to the audit2why problem similar to the bellow one.

**Comment by vmojzis** [comment]
Updated with os._exit() and additional comments.

File: framework/src/setroubleshoot/server.py [hunk 2]
```diff
@@ -115,6 +115,13 @@ def sighandler(signum, frame):
         return
 
 
+def polling_failled_handler(signum, frame):
```
Comment (janzarsky) [inline framework/src/setroubleshoot/server.py]: there is a typo, it should probably be 'polling_failed_handler'

Comment (vmojzis) [reply framework/src/setroubleshoot/server.py]: Thanks.

**Pull request merged**